packages/ethers/src/provider-adapter.ts

```diff
@@ -13,30 +13,25 @@ import {
 import { defineReadOnly } from "@ethersproject/properties";
 import { JsonRpcProvider } from "@ethersproject/providers";
 import { AccountSigner } from "./account-signer.js";
-
-export type EthersProviderAdapterOpts = { entryPointAddress?: Address } & (
-  | {
-      rpcProvider: string | PublicErc4337Client<HttpTransport>;
-      chainId: number;
-    }
-  | {
-      accountProvider: SmartAccountProvider<HttpTransport>;
-    }
-);
+import { EthersProviderAdapterOptsSchema } from "./schema.js";
+import type { EthersProviderAdapterOpts } from "./types.js";
 
 /** Lightweight Adapter for SmartAccountProvider to enable Signer Creation */
 export class EthersProviderAdapter extends JsonRpcProvider {
   readonly accountProvider: SmartAccountProvider<HttpTransport>;
 
   constructor(opts: EthersProviderAdapterOpts) {
+    EthersProviderAdapterOptsSchema.parse(opts);
+
     super();
     if ("accountProvider" in opts) {
       this.accountProvider = opts.accountProvider;
     } else {
       const chain = getChain(opts.chainId);
       this.accountProvider = new SmartAccountProvider({
         rpcProvider: opts.rpcProvider,
-        entryPointAddress: opts.entryPointAddress,
+        entryPointAddress:
+          opts.entryPointAddress ?? getDefaultEntryPointAddress(chain),
```

Comment:
this should be reverted. 

From PR https://github.com/alchemyplatform/aa-sdk/pull/181/

```
// sanity check. Note that this check is only performed iff the optional entryPointAddress is given upon initialization.
    if (
      this.entryPointAddress &&
      account.getEntryPointAddress() !== this.entryPointAddress
    ) {
      throw new Error(
        `Account entryPoint address: ${account.getEntryPointAddress()} does not match the current provider's entryPoint address: ${
          this.entryPointAddress
        }`
      );
    }